File: python/template/solution_test.py
from solution import part_1, part_2

##################
# Helper Methods #
##################


def get_input(filename: str):
    file = open(filename)
    return file.read()


def run_test(fun, input: str, want: int):
    got = fun(input)
    assert got == want


#########
# Tests #
#########


def test_part_1():
    inputs = {
        "sample1": 0,
        "sample2": 0,
        "sample3": 0,
    }

    for k, v in inputs.items():
        run_test(part_1, k, v)


def test_part_2():
    inputs = {
        "sample1": 0,
        "sample2": 0,
        "sample3": 0,
    }

    for k, v in inputs.items():
        run_test(part_2, k, v)

File: go/template/solution_test.go
package solution

import "testing"

// Optionally convert these to table-driven tests
func TestPart1(t *testing.T) {
	t.Run("sample input", func(t *testing.T) {
		got := Part1()
		want := 0

		assertSolution(t, got, want)
	})
	t.Run("actual input", func(t *testing.T) {
		got := Part1()
		want := 0

		assertSolution(t, got, want)
	})
}

func TestPart2(t *testing.T) {
	t.Run("sample input", func(t *testing.T) {
		got := Part2()
		want := 1

		assertSolution(t, got, want)
	})
	t.Run("actual input", func(t *testing.T) {
		got := Part2()
		want := 1

		assertSolution(t, got, want)
	})
}

func assertSolution(t testing.TB, got int, want int) {
	t.Helper()

	if got != want {
		t.Errorf("got %d, wanted %d", got, want)
	}
}
